storage/storage.py

```diff
@@ -84,6 +106,42 @@ def load_from_file(self, in_file):
     def delete(self):
         os.remove(self._parsed_storage_uri.path)
 
+    def get_download_url(self, seconds=60, key=None):
+        """
+        Return a temporary URL allowing access to the storage object.
+
+        If a download_url_base is specified in the storage URI, then a call to get_download_url() will
+        return the download_url_base joined with the object name.
+
+        For example, if "http://www.someserver.com:1234/path/to/" were passed (urlencoded) as the
+        download_url_base query parameter of the storage URI:
+
+          file://some/path/to/a/file.txt?download_url_base=http%3A%2F%2Fwww.someserver.com%3A1234%2Fpath%2Fto%2
+
+        then a call to get_download_url() would yield:
+
+          http://www.someserver.com:1234/path/to/file.txt
+
+
+        :param seconds: ignored for local storage
+        :param key:     ignored for local storage
+        :return:        the download url that can be used to access the storage object
+        :raises:        DownloadUrlBaseUndefinedError
+        """
+        return generate_download_url_from_base(self._download_url_base,
+                self._parsed_storage_uri.path.split('/')[-1])
+
+
+def generate_download_url_from_base(base, object):
```

Comment:
Maybe say `object_name`, since it's just the name, not the literal `StorageObject` instance (plus, `object` is a reserved name that you're redefining).`